amf-api-contract/shared/src/main/scala/amf/apicontract/internal/spec/raml/parser/context/RamlSpecAwareContext.scala

```diff
@@ -40,7 +40,7 @@ abstract class RamlSpecVersionFactory(implicit val ctx: RamlWebApiContext) exten
 
   def documentParser: (Root) => RamlDocumentParser
 
-  def typeParser: (YMapEntry, Shape => Unit, Boolean, DefaultType) => RamlTypeParser
+  def typeParser(entry: YMapEntry, adopt: Shape => Unit, isAnnotation: Boolean, default: DefaultType): RamlTypeParser
```

Comment:
My point is that we were working with functions as return type, and now is returning the parser but just for types. All other methods at web api level keep returning functions to build parsers, not parsers it self

Reply:
The fact that we return functions to build parsers in this context doesn't make sense. Why ? because we never curry them or pass them as parameters elsewhere, we just call them on the spot.

The way we use them, returning a function that build the parser instead of having a function that returns the parser is just syntax sugar. In fact, is is more clear to have a function to return a parser because of the parameter names. If the developer doesn't which params the "function that returns a function that builds a parser works" then he/she would have to scout the uses to read it.

Reply:
In fact, as functions are first class values in scala this is just syntax sugar as it is the same to define the function or to return a function as in either case someone can still grab the function by value. For example:

```scala

class X {
  def a(a: String): String = { // something}
  
  val function = a // this works
  val result = a("something")
  
  def b: String => String = { // something }
  val result = b("something")
  
  def receivesFunc(func: String => String)
  
  receivesFunc(a) // works
  receivesFunc(b) // works
}
```

Reply:
I'm not arguing if one is better from the other. My point is that all this builders returns functions, and this chance is breaking that pattern. That will be more confusing than the function.

Reply:
I changed it.